### PR TITLE
v3

### PR DESCRIPTION
**Pull request by rasmusbe**
The one that breaks stuff

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,55 +1,94 @@
...
+| Method | Arguments                                                      | Returns  |
+| -------|:---------------------------------------------------------------|---------:|
+| parse  | string personnummer, [ array options<sup>[*](#options)</sup> ] | Instance |
+| valid  | string personnummer, [ array options<sup>[*](#options)</sup> ] | bool     |
+
+#### Instance
+| Method               | Arguments       | Returns |
+| ---------------------|:----------------|--------:|
+| format               | bool longFormat | string  |
+| getAge               | none            | int     |
```
Comment (frozzare) [inline README.md]: `getAge` will be removed in v3 and replaced with `age` property

**Comment by frozzare** [comment]
LGTM, @Johannestegner any input?

File: src/Personnummer.php [hunk 2]
```diff
@@ -1,52 +1,70 @@
...
+     * @param string $ssn
+     * @param array  $options
      *
-     * @return bool
+     * @return PersonnummerInterface
+     *
+     * @throws PersonnummerException
      */
-    public static function valid($ssn, $includeCoordinationNumber = true)
+    public static function parse($ssn, $options = []): PersonnummerInterface
```
Comment (Johannestegner) [inline src/Personnummer.php]: We could strong type the `$ssn` param to string (`string $ssn`) to make php typecheck it.

File: src/Personnummer.php [hunk 3]
```diff
@@ -85,77 +97,24 @@ public static function format($ssn, $longFormat = false)
...
-     * Check if a Swedish social security number is for a male.
-     *
-     * @param string|int $ssn                       Social security number to test.
-     * @param bool       $includeCoordinationNumber If to include Coordination numbers as valid numbers.
-     *
-     * @return bool
-     * @throws PersonnummerException On parse failure.
-     */
-    public static function isMale($ssn, $includeCoordinationNumber = true)
+    public static function valid($ssn, array $options = []): bool
```
Comment (Johannestegner) [inline src/Personnummer.php]: As above, strong type on `$ssn` would make the type-error catch not needed.

File: README.md (hunk 1 above)
Comment (Johannestegner) [reply README.md]: Seeing php only have properties in form of the magic `__get` method or through public methods, I think that its okay to have both a `getAge` and a magic `__get` for the properties.

I'm fine with it staying, whilst I wouldn't be angry with it gone :P

Comment (rasmusbe) [reply README.md]: I agree with @Johannestegner. No need to remove it for now.

Comment (frozzare) [reply README.md]: I thought it was the old `getAge` method and `age` is missing from the property list :)

Comment (Johannestegner) [reply README.md]: Ahh, okay, I missed that one. Could you add that @rasmusbe?

**Pull request merged**